packages/sdk/src/evm/wallets/core/aws.ts

```diff
@@ -0,0 +1,79 @@
+import { AbstractWallet } from "../types/abstract";
+import {
+  GetSecretValueCommand,
+  SecretsManagerClient,
+  SecretsManagerClientConfig,
+} from "@aws-sdk/client-secrets-manager";
+import { ethers } from "ethers";
+
+export type AwsSecretsManagerWalletOptions = {
+  secretId: string;
+  secretKeyName: string;
+  awsConfig: SecretsManagerClientConfig;
+};
+
+/**
+ * Create a wallet instance using a private key stored in AWS Secrets Manager.
+ *
+ *  @example
+ * ```javascript
+ * import { ThirdwebSDK } from "@thirdweb-dev/sdk"
+ * import { AWSSecretsManagerWallet } from "@thirdweb-dev/sdk/evm/wallets"
+ *
+ * const wallet = new AwsSecretsManagerWallet({
+ *   secretName: "my-secret",
+ *   secretKeyName: "private-key",
+ *   awsConfig: {
+ *     region: "us-east-1",
+ *     credentials: {
+ *       accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+ *       secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+ *     },
+ *   }
+ * });
+ *
+ * const sdk = await ThirdwebSDK.fromWallet(wallet, "mainnet");
+ * ```
+ */
+export class AWSSecretsManagerWallet extends AbstractWallet {
+  private secretId: string;
+  private secretKeyName: string;
+  private client: SecretsManagerClient;
+  private cachedSigner: ethers.Signer | undefined;
+
+  constructor(options: AwsSecretsManagerWalletOptions) {
+    super();
```

Comment:
does `AbstractWallet` have a constructor? if not this is not needed

Reply:
No but it still yells at you to always call super for some reason